cli/cycpp.py

```diff
@@ -1045,224 +1084,236 @@ def _val(self, t, val=None, name=None, uitype=None):
+    def _val_int(self, t, val=0, name='intvar', uitype=None):
         if uitype == 'nuclide':
             fmt = '"{0}"' if isinstance(val, STRING_TYPES) else '{0}'
             v = fmt.format(val)
-            v = 'pyne::nucname::id({0})'.format(v)
+            v = 'int {0} = pyne::nucname::id({1});'.format(name, v)
         else:
-            v = '{0}'.format(val)
+            v = 'int {0} = {1};'.format(name, val)
         return v
 
-    def _val_floating(self, t, val=0.0, name=None, uitype=None):
-        return '{0}'.format(val)
+    def _val_floating(self, t, val=0.0, name='floatvar', uitype=None):
+        return 'double {0} = {1};'.format(name, val)
 
-    def _val_string(self, t, val='', name=None, uitype=None):
-        return 'std::string("{0}")'.format(val)
+    def _val_string(self, t, val='', name='stringvar', uitype=None):
+        return 'std::string {0}("{1}");'.format(name, val)
 
-    def _val_blob(self, t, val='', name=None, uitype=None):
-        return 'cyclus::Blob("{0}")'.format(val)
+    def _val_blob(self, t, val='', name='blobvar', uitype=None):
+        return 'cyclus::Blob {0}("{1}");'.format(name, val)
 
-    def _val_uuid(self, t, val='', name=None, uitype=None):
+    def _val_uuid(self, t, val='', name='uuidvar', uitype=None):
+        v = 'boost::uuids::uuid {0} = '.format(name)
         if isinstance(val, STRING_TYPES):
-            v = '"{0}"'.format(val)
+            v += '"{0}"'.format(val)
         elif isinstance(val, uuid.UUID):
-            v = [x+y for x,y in zip(val.hex[::1], val.hex[1::2])]
-            v = '{0x' + ', 0x'.join(v) + '}'
+            l = [x+y for x,y in zip(val.hex[::1], val.hex[1::2])]
+            v += '{0x' + ', 0x'.join(l) + '}'
         else:
             msg = "could not interpret UUID type of {0}"
             raise TypeError(msg.format(val))
-        return v
+        return v + ';'
+
+    def _val_vector(self, t, val=(), name='vecvar', uitype=None):
+        uitype = prepare_type(t, uitype)
+        elemname = 'elem'
 
-    def _val_vector(self, t, val=(), name=None, uitype=None):
-        vtype = t[1]
-        vuitype = None if uitype is None else uitype[1]
-        v = '{0}.resize({1});\n'.format(name, len(val))
+        v = '{0} {1};'.format(type_to_str(t), name)
+        v += '{0}.resize({1});\n'.format(name, len(val))
+        v += '{'
         for i, x in enumerate(val):
-            x = self._val(vtype, val=x, uitype=vuitype)
-            v += '{0}[{1}] = {2};\n'.format(name, i, x)
+            x = self._val(t[1], val=x, uitype=uitype[1], name=elemname)
+            v += '{{ {0} {1}[{2}] = {3};}}'.format(x, name, i, elemname)
+        v += '}'
+
         return v
 
-    def _val_set(self, t, val=frozenset(), name=None, uitype=None):
-        vtype = t[1]
-        vuitype = None if uitype is None else uitype[1]
-        v = ''
+    def _val_set(self, t, val=frozenset(), name='setvar', uitype=None):
+        uitype = prepare_type(t, uitype)
+        elemname = 'elem'
```

Comment:
This is in InfileToDb - and this variable name is referring to a temporary c++ var - nothing to do with schema tag names.